titus-api/src/main/java/com/netflix/titus/api/jobmanager/model/job/ebs/EbsVolumeUtils.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Copyright 2020 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netflix.titus.api.jobmanager.model.job.ebs;
+
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.netflix.titus.api.jobmanager.JobAttributes;
+import com.netflix.titus.api.jobmanager.model.job.Job;
+import com.netflix.titus.api.jobmanager.model.job.JobDescriptor;
+import com.netflix.titus.api.jobmanager.model.job.Task;
+import com.netflix.titus.common.util.StringExt;
+
+import static com.netflix.titus.api.jobmanager.TaskAttributes.TASK_ATTRIBUTES_EBS_VOLUME_ID;
+
+/**
+ * Helper utilities for processing EBS volume related info.
+ */
+public class EbsVolumeUtils {
+
+    public static <E extends JobDescriptor.JobDescriptorExt> List<EbsVolume> getEbsVolumes(JobDescriptor<E> jobDescriptor) {
+        List<String> volumeIds = getVolumeIds(jobDescriptor);
+        Optional<String> mountPointOptional = getMountPoint(jobDescriptor);
+        Optional<EbsVolume.MountPerm> mountPermOptional = getEbsMountPerm(jobDescriptor);
+        Optional<String> fsTypeOptional = getEbsFsType(jobDescriptor);
+
+        if (!(mountPointOptional.isPresent() && mountPermOptional.isPresent() && fsTypeOptional.isPresent())) {
+            return Collections.emptyList();
+        }
+
+        String mountPoint = mountPointOptional.get();
+        EbsVolume.MountPerm mountPerm = mountPermOptional.get();
+        String fsType = fsTypeOptional.get();
+
+        return volumeIds.stream()
+                .map(volumeId -> EbsVolume.newBuilder()
```

Comment:
The null and 0 values would indicate that those values have not yet been set. Currently, the expectation is that subsequent model validation will catch those not being set. Did you have something in mind for how to mark them?

Reply:
Nothing concrete. I was just thinking what would be the best way to deal with partially constructed object.
In general, we do not pass any null values in core model to avoid NPEs.